Gems/Multiplayer/Code/Tests/SimplePlayerSpawnerTests.cpp

```diff
@@ -0,0 +1,129 @@
+        {
+            auto spawner = entity.CreateComponent<SimplePlayerSpawnerComponent>();
+            spawner->GetSpawnPoints() = spawnPoints;
+            entity.Init();
+            entity.Activate();
+        }
+
+        void CreateSpawnPoint(AZ::Entity& spawnPointEntity, const AZ::Vector3& position)
+        {
+            auto transform = spawnPointEntity.CreateComponent<AzFramework::TransformComponent>();
+            transform->SetWorldTM(AZ::Transform::CreateTranslation(position));
+
+            spawnPointEntity.Init();
+            spawnPointEntity.Activate();
+        }
+
+        AZStd::unique_ptr<BenchmarkComponentApplicationRequests> m_componentApplicationRequests;
+        AZStd::unique_ptr<AZ::SerializeContext> m_serializeContext;
+        AZStd::unique_ptr<AZ::ComponentDescriptor> m_transformDescriptor;
+        AZStd::unique_ptr<AZ::ComponentDescriptor> m_simplePlayerSpawnerComponentDescriptor;
+    };
+
+    TEST_F(SimplePlayerSpawnerTests, RoundRobinWrapAround)
+    {
+        AZ::Entity spawnPoint1(AZ::EntityId(1));
+        AZ::Entity spawnPoint2(AZ::EntityId(2));
+        AZ::Entity spawnPoint3(AZ::EntityId(3));
+
+        CreateSpawnPoint(spawnPoint1, AZ::Vector3(1.0f, 0, 0));
+        CreateSpawnPoint(spawnPoint2, AZ::Vector3(2.0f, 0, 0));
+        CreateSpawnPoint(spawnPoint3, AZ::Vector3(3.0f, 0, 0));
+
+        AZ::Entity simplePlayerSpawnerEntity;
+        CreateSimplePlayerSpawner(simplePlayerSpawnerEntity, { spawnPoint1.GetId(), spawnPoint2.GetId(), spawnPoint3.GetId() });
+
+        uint32_t numSpawnPoints;
+        SimplePlayerSpawnerRequestBus::BroadcastResult(numSpawnPoints, &SimplePlayerSpawnerRequests::GetSpawnPointCount);
+        EXPECT_EQ(numSpawnPoints, 3);
+
+        AZ::Transform nextSpawnPoint;
+        SimplePlayerSpawnerRequestBus::BroadcastResult(nextSpawnPoint, &SimplePlayerSpawnerRequests::RoundRobinNextSpawnPoint);
+        EXPECT_EQ(nextSpawnPoint.GetTranslation().GetX(), 1);
+
+        SimplePlayerSpawnerRequestBus::BroadcastResult(nextSpawnPoint, &SimplePlayerSpawnerRequests::RoundRobinNextSpawnPoint);
+        EXPECT_EQ(nextSpawnPoint.GetTranslation().GetX(), 2);
+
+        SimplePlayerSpawnerRequestBus::BroadcastResult(nextSpawnPoint, &SimplePlayerSpawnerRequests::RoundRobinNextSpawnPoint);
+        EXPECT_EQ(nextSpawnPoint.GetTranslation().GetX(), 3);
+
+        // Ran through each spawn point; make sure the spawner goes back to spawn point #1
+        SimplePlayerSpawnerRequestBus::BroadcastResult(nextSpawnPoint, &SimplePlayerSpawnerRequests::RoundRobinNextSpawnPoint);
+        EXPECT_EQ(nextSpawnPoint.GetTranslation().GetX(), 1);
+
+        AZ::Outcome<uint32_t, AZStd::string> spawnPointIndexOutcome;
+        SimplePlayerSpawnerRequestBus::BroadcastResult(spawnPointIndexOutcome, &SimplePlayerSpawnerRequests::GetNextSpawnPointIndex);
+        EXPECT_TRUE(spawnPointIndexOutcome.IsSuccess());
+        EXPECT_EQ(spawnPointIndexOutcome.GetValue(), 1); // Spawn point #2 (aka index 1)
```

Comment:
Nit: It's _really_ not important, but consider making the constants floats (1.0f, 2.0f, etc) since that's what you're actually comparing here.

Reply:
good call! especially easier on the eyes to differentiate the other numbers like when checking indices. Much clearer to read. Thanks!